### ISSUE TITLE
Death End Re;Quest 2 (1266220)

### ISSUE THREAD

**Issue by CygnusTerminal** [ValveSoftware/Proton#4135]
# Compatibility Report
- Name of the game with compatibility issues: Death End Re;Quest 2
- Steam AppID of the game: 1266220

## System Information
- GPU: RX 480
- Driver/LLVM version: Mesa 20.1.5
- Kernel version: 5.8
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/CygnusTerminal/d440e28b80de63d9fa0d09257f16f270
- Proton version: 5.0-9

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

[steam-1266220.log](https://github.com/ValveSoftware/Proton/files/5092381/steam-1266220.log)

## Symptoms 
Game wont run, DX11 error appears and it inmediatly closes (the text is gibberish)

## Reproduction
Launch the game
See the error

**Comment by CygnusTerminal** [comment]
[steam-1266220.log](https://github.com/ValveSoftware/Proton/files/5092436/steam-1266220.log)
Proton-5.9-GE-5-ST manages to open the game with a black screen and it instantly closes.

**Comment by kisak-valve** [comment]
Hello @CygnusTerminal, `fixme:vulkan:X11DRV_vkCreateWin32SurfaceKHR Application requires child window rendering, which is not implemented yet!` in the log points towards https://bugs.winehq.org/show_bug.cgi?id=45277.

**Comment by CygnusTerminal** [comment]
> 
> 
> Hello @CygnusTerminal, `fixme:vulkan:X11DRV_vkCreateWin32SurfaceKHR Application requires child window rendering, which is not implemented yet!` in the log points towards https://bugs.winehq.org/show_bug.cgi?id=45277.

Oh, well, should i post this on the dxvk repo then?

**Comment by kisak-valve** [comment]
My understanding is that winevulkan is responsible for that in upstream wine and the linked bug report is the right place to discuss it.

**Comment by CygnusTerminal** [comment]
> 
> 
> My understanding is that winevulkan is responsible for that in upstream wine and the linked bug report is the right place to discuss it.

Oh, sorry, thanks for the heads up kisak! I'll try the latest wine with those patches discussed in the thread and see if it helps.

**Comment by CygnusTerminal** [comment]
Compiled proton-tkg from scratch with child window patch, still opens a windows and instantly closes. Log is attached, lines of interest:
wine: Unhandled page fault on write access to 0000000000000001 at address 000000007B63DE0A (thread 00c0), starting debugger..
Unhandled exception: page fault on write access to 0x00000001 in 64-bit code (0x000000007b63de0a).
[steam-1266220.log](https://github.com/ValveSoftware/Proton/files/5103063/steam-1266220.log)

**Comment by kisak-valve** [comment]
Looks like this game would benefit from support for XACT Engine maturing (`fixme:xact3:IXACT3EngineImpl_Initialize pXAudio2 parameter not supported! Falling back to NULL`).

**Comment by CygnusTerminal** [comment]
> Looks like this game would benefit from support for XACT Engine maturing (`fixme:xact3:IXACT3EngineImpl_Initialize pXAudio2 parameter not supported! Falling back to NULL`).

Yeah, it would seem so, sorry to bother kisak, would installing XACT with winetricks and setting the dll overrides help?

**Comment by ra59674** [comment]
> > Looks like this game would benefit from support for XACT Engine maturing (`fixme:xact3:IXACT3EngineImpl_Initialize pXAudio2 parameter not supported! Falling back to NULL`).
> 
> Yeah, it would seem so, sorry to bother kisak, would installing XACT with winetricks and setting the dll overrides help?

I managed to get this game working by installing XACT with winetricks, I also needed to add the following overrides xaudio2_7 and xactengine3_7. The game should run flawlessly with this setup and Proton-5.9-GE-5-ST.

**Comment by CygnusTerminal** [comment]
> > > Looks like this game would benefit from support for XACT Engine maturing (`fixme:xact3:IXACT3EngineImpl_Initialize pXAudio2 parameter not supported! Falling back to NULL`).
> > 
> > 
> > Yeah, it would seem so, sorry to bother kisak, would installing XACT with winetricks and setting the dll overrides help?
> 
> I managed to get this game working by installing XACT with winetricks, I also needed to add the following overrides xaudio2_7 and xactengine3_7. The game should run flawlessly with this setup and Proton-5.9-GE-5-ST.

Nice find! I managed to get it working yesterday with the same after tinkering a bit, but forgot to update

**Comment by Naitrate** [comment]
I was able to get it up and running using GloriousEggroll's fork of Proton, alongside using ProtonTricks to set up cinepack, faudio, and xact_x64 (The things that someone on ProtonDB recommended for Megadimension Neptunia VIIR).

However, It's entirely possible that this configuration causes extreme stuttering when moving the mouse around, basically mandating a controller to be used to stop said stuttering. I'm wondering if anyone else has that issue with the mouse.

**Comment by CygnusTerminal** [comment]
> I was able to get it up and running using GloriousEggroll's fork of Proton, alongside using ProtonTricks to set up cinepack, faudio, and xact_x64 (The things that someone on ProtonDB recommended for Megadimension Neptunia VIIR).
> 
> However, It's entirely possible that this configuration causes extreme stuttering when moving the mouse around, basically mandating a controller to be used to stop said stuttering. I'm wondering if anyone else has that issue with the mouse.

Nice to see another take on how to make it work, i'll try the xact fix with overrides first and see if mouse does not lag+if everything works (sound, cinematics, etc) and report back.

**Comment by CygnusTerminal** [comment]
UPDATE: Game runs perfectly fine with proton-ge, my custom proton-tkg was causing mouse pointer issues (appearing and dissapearing on screen). Also, for AMD users running aco, i found some graphical glitches (textures fading to black on a certain camera position). Finally, i'll test, but i'm getting some performance issues compared to my Windows 10 install, but i have the game on my NVMe on Windows, and in Linux the game is installed to my HDD, so further testing needs to be done.

**Comment by kisak-valve** [comment]
Hello @CygnusTerminal, is RADV/llvm equally affected? It would be nice to have a screenshot of the graphical quirk and confirmation that mesa 20.2 or mesa git-master are also affected.

**Comment by CygnusTerminal** [comment]
> Hello @CygnusTerminal, is RADV/llvm equally affected? It would be nice to have a screenshot of the graphical quirk and confirmation that mesa 20.2 or mesa git-master are also affected.

Hello kisak! As far as i tested, llvm is not affected by said graphical glitch. I'm working on getting some footage of it, and will upload ASAP.

**Comment by CygnusTerminal** [comment]
@kisak-valve https://drive.google.com/file/d/1Y81G2wU6GRWMITR8LqA1c_tUnCuTMlSe/view?usp=sharing There you go, Those are the videos with the glitches that appear when running aco as RADV's compiler.
Edit: Re-testing llvm more thoroughly as we speak, also affected, seems to be a Mesa bug. I'll test later with my nvidia laptop and see if it suffers from the same. Windows does not show this behaviour

**Comment by hakzsam** [comment]
@CygnusTerminal Do you still have issues with ACO?

**Comment by CygnusTerminal** [comment]
> @CygnusTerminal Do you still have issues with ACO?

Hi, i'll try this evening and report back.

**Comment by CygnusTerminal** [comment]
Reporting back: Issues seems to be gone, there are no longer graphical glitches, but it seems Proton Experimental introduced audio cracking. Will try with the latest good known fork i used and see if it's a problem with my PC or Proton.

**Comment by CygnusTerminal** [comment]
[steam-1266220.log](https://github.com/ValveSoftware/Proton/files/5756196/steam-1266220.log)
Well, sadly, Proton is now showing audio cracking, which was not present last time i played, this only happens within Proton, and depending on the version, it appears quite fast. For instance, Proton Experimental shows cracking inmediatly after i load a savegame. It's very loud and noticiable, to the point of affecting the game. Proton-GE 5.9 ST 5 gets progressively louder the more you play. This is in Arch, with pulseaudio and the default settings, on kernel 5.10 (tkg-pds). Will try with the vanilla kernel to see if it helps.
Edit: Vanilla kernel still shows the same problem.

**Comment by hakzsam** [comment]
Thanks for checking!

**Comment by CygnusTerminal** [comment]
> Thanks for checking!

No problem! I do commend you to try, since i think the sound issues come from my own mistake (being forced to lurk on Windows for the last 2 months didn't help). I'll reinstall since i am also having weird problems like FPS drops on the menu, and the popping noise.

**Comment by trenton4k** [comment]
## System Information
- GPU: rx 6800
- Driver/LLVM version: Mesa 22.0.0-devel
- Kernel version: 5.14
- Proton version:1636974995 experimental-bleeding
[steam-1266220.log](https://github.com/ValveSoftware/Proton/files/7540041/steam-1266220.log)

game will not launch

**Comment by kisak-valve** [comment]
Hello @trenton4k, these look like the lines of interest from your log:
```
err:module:import_dll Library steam_api64.dll (which is needed by L"Z:\\mnt\\86453d71-9bee-487c-82b1-868b9be0d595\\SteamLibrary\\steamapps\\common\\DeathEndReQuest2\\resource\\bin\\DeathEndReQuest2.exe") not found
[...]
err:module:LdrInitializeThunk Importing dlls for L"Z:\\mnt\\86453d71-9bee-487c-82b1-868b9be0d595\\SteamLibrary\\steamapps\\common\\DeathEndReQuest2\\resource\\bin\\DeathEndReQuest2.exe" failed, status c0000135
```
It looks like `<game folder>/resource/bin/steam_api64.dll` should be provided by depot 1266221.

Please verify the integrity of the game's files in Steam.
